Compatibility/MainWindow.xaml.cs

```diff
@@ -0,0 +1,301 @@
+                    tCompatPref = Assembly.Load("PresentationCore").GetType($"System.Windows.CoreCompatibilityPreferences");
+                    break;
+                case "FrameworkCompatibilityPreferences":
+                    tCompatPref = Assembly.Load("PresentationFramework").GetType($"System.Windows.FrameworkCompatibilityPreferences");
+                    break;
+
+            }
+            return tCompatPref;
+        }
+
+        public static bool TryGetAppContextSwitchValue(string className, string switchName, out bool switchValue)
+        {
+            switchValue = false;
+
+            Type tSwitch = GetAppContextSwitchType(className);
+
+            if (tSwitch == null)
+            {
+                return false;
+            }
+
+            var pProperty = tSwitch.GetProperty(switchName, BindingFlags.Static | BindingFlags.Public);
+            if (pProperty != null)
+            {
+                try
+                {
+                    switchValue = (bool)pProperty.GetValue(null);
+                    return true;
+                }
+                catch { }
+            }
+
+            return false;
+        }
+
+        private bool TryGetCompatibilityPreferenceSwitchValue(string compatPreferenceClass, string switchName, out bool switchValue)
+        {
+            switchValue = false;
+            Type tSwitch = GetCompatPreferenceType(compatPreferenceClass);
+            if (tSwitch == null)
+            {
+                return false;
+            }
+
+            var pProperty = tSwitch.GetProperty(switchName, BindingFlags.Static | BindingFlags.Public);
+            if (pProperty == null)
+            {
+                pProperty = tSwitch.GetProperty(switchName, BindingFlags.Static | BindingFlags.NonPublic);
+            }
+
+            if (pProperty != null)
+            {
+                try
+                {
+                    switchValue = (bool)pProperty.GetValue(null);
+                    return true;
+                }
+                catch
+                {
+                }
+            }
+
+            return false;
+        }
+
+        private bool TryGetCompatibilityPreferenceSwitchValue(string compatPreferenceClass, string switchName, out string switchString)
+        {
+            switchString = null;
+            Type tSwitch = GetCompatPreferenceType(compatPreferenceClass);
+            if (tSwitch == null)
+            {
+                return false;
+            }
+
+            var pProperty = tSwitch.GetProperty(switchName, BindingFlags.Static | BindingFlags.Public);
+            if (pProperty == null)
+            {
+                pProperty = tSwitch.GetProperty(switchName, BindingFlags.Static | BindingFlags.NonPublic);
+            }
+
+            if (pProperty != null)
+            {
+                try
+                {
+                    object o = pProperty.GetValue(null);
+                    if (pProperty.PropertyType == typeof(bool?))
```

Comment:
Could you use `is` to replace it?

```
if ( o is bool? b)
```

Reply:
no, not here. `object o` could be `null`.